src/App.tsx

```diff
@@ -15,48 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ByLength = 'length',
+  ByAlphabet = 'alphabet',
+  Default = '',
+}
+
+function sortBy(sortType: SortType, isReversed: boolean): string[] {
+  const goods = [...goodsFromServer];
+
+  if (sortType) {
+    switch (sortType) {
+      case SortType.ByAlphabet:
+        goods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SortType.ByLength:
+        goods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.Default);
+  const shouldShowResetButton = sortType || isReversed;
+
+  const preparedGoods: string[] = sortBy(sortType, isReversed);
```

Comment:
I think it is unnecessary here because typescript already know it's a `string[]`

```suggestion
  const preparedGoods = sortBy(sortType, isReversed);
```